### PR TITLE
added edge tests

### PR DESCRIPTION
**Pull request by nickjsr**
I added some edge tests for your review. Thanks

### CODE REVIEW

**Comment by nickjsr** [comment]
This version has many lines and many local variables according to pylint report. We can fix them in the end.

File: test/pipeline_test.py [hunk 1]
```diff
@@ -44,26 +44,74 @@ def test_drop():
 
     # Drop by columns
     df1 = DplyFrame(pandas_df)
-    output1 = df1 + drop(["col3", "col4"], axis=1)
-    expected1 = pandas_df.drop(["col3", "col4"], axis=1)
+    output1 = df1 + drop("col1", axis=1)
+    expected1 = pandas_df.drop("col1", axis=1)
```
Comment (MatthewChau99) [inline test/pipeline_test.py:47]: For this part, I was actually testing with different input parameters. First I was trying to use the positional arg (plus the axis), and then the columns keyword arg (without the axis arg), so I think it's better to keep it the way it was before.

File: test/pipeline_test.py [hunk 2]
```diff
@@ -44,26 +44,74 @@ def test_drop():
...
-    output3 = df3 + drop([0, 1])
-    expected3 = pandas_df.drop([0, 1])
+    output3 = df3 + drop(columns="col1")
+    expected3 = pandas_df.drop(columns="col1")
     pd.testing.assert_frame_equal(output3.pandas_df, expected3)
 
     df4 = DplyFrame(pandas_df)
-    output4 = df4 + drop(index=[2, 3]) + drop(index=[1])
-    expected4 = pandas_df.drop(index=[1, 2, 3])
+    output4 = df4 + drop(columns=["col3", "col4"])
```
Comment (MatthewChau99) [inline test/pipeline_test.py:68]: Same here - trying to test all of the arguments.

**Comment by nickjsr** [comment]
I have updated my work for your review. Thanks!

**Review by MatthewChau99 — APPROVED** [review]
Looks a lot nicer.

File: test/pipeline_test.py (hunk 1 above)
Comment (nickjsr) [reply test/pipeline_test.py:47]: My output2 covers your previous output1

File: test/pipeline_test.py (hunk 2 above)
Comment (nickjsr) [reply test/pipeline_test.py:68]: Your outpu4 is my output8

**Comment by nickjsr** [comment]
My new push dealt with dplypy_test.py. For pipeline_test.py, I am definitely sure that I cover all previous tests you wrote, all of which are perfect. Based on the specific parameters being tested, I renamed lots of tests, which made the file changed messy.

File: test/pipeline_test.py (hunk 2 above)
Comment (MatthewChau99) [reply test/pipeline_test.py:68]: Oh sorry I missed that. Thanks for reminding me!

**Review by MatthewChau99 — APPROVED** [review]

**Review by ccharp — APPROVED** [review]

**Pull request merged**